skopt/optimizer/optimizer.py

```diff
@@ -195,6 +196,7 @@ def tell(self, x, y, fit=True):
             only be fitted after `n_random_starts` points have been queried
             irrespective of the value of `fit`.
         """
+        is_categorical = all([isinstance(dim, Categorical) for dim in self.space.dimensions])
```

Comment:
Shouldnt this be set once at init instead? Or better, as propery of the `space` instance?